### PR TITLE
Add search 'by-path' url for browser

### PR DESCRIPTION
**Pull request by biquadrates-artsman**
This PR adds the option to use the URL scheme to search for an entry by path. Hence, tools implementing the keepassxc-browser API can use a credential's path to request a specific credential.

The code currently reuses the strategy implemented by @piegamesde in https://github.com/keepassxreboot/keepassxc/pull/4763.

## Testing strategy
- Unit tests.

## Type of change
- ✅ New feature (change that adds functionality)

### CODE REVIEW

**Comment by marquardts** [comment]
As the path is known, I think one should avoid the search. What do you think @piegamesde ?

File: src/browser/BrowserService.cpp
Comment (piegamesde) [inline src/browser/BrowserService.cpp:1020]: I'm really not sure, but could this method be added as method of `Entry` instead? It feels like it would be better to put it there.

**Review by piegamesde — COMMENTED** [review]
Wow, that was quick! I don't see any problems, but also I'm not really familiar with the code base of keepassxc

**Comment by droidmonkey** [comment]
I would rather we implement the proposed reference syntax from #5369

**Comment by marquardts** [comment]
Yes, you are right.In case that the proposal is the answer, I can adapt my change proposal accordingly, as well as refactor the by-uuid syntax. The question would be, if you already require the full implementation of the syntax? E.g. retrieving history items etc via the browser interface.

**Comment by droidmonkey** [comment]
We can merge this, once fixed up, for now and easily replace it with the new syntax.

File: src/core/Entry.cpp
Comment (marquardts) [inline src/core/Entry.cpp:515]: @droidmonkey , this is executed with each request to `path()`. It can be refactored into a property but then updating etc would also need to be implemented. I do not believe that this operation is _that_ costly to justify the amount of extra code otherwise needed.

Comment (droidmonkey) [reply src/core/Entry.cpp:515]: We have a lot of lookup inefficiencies i am investigating for 2.7, especially related to searching.

**Comment by marquardts** [comment]
As far as I am concerned, this PR is ready for review.

**Comment by marquardts** [comment]
@droidmonkey, @piegamesde Do I need to add something here?

**Comment by piegamesde** [comment]
The changes are fine for me, but I can't judge how this interacts with #5369.

**Comment by marquardts** [comment]
> The changes are fine for me, but I can't judge how this interacts with #5369.

Well, once #5369 is implemented, we would need to adjust the URL here. See https://github.com/keepassxreboot/keepassxc/pull/5535#issuecomment-706692691

Of course, IMO, the implementation of #5369 would imply the implementation of a generic search handler such that the bells-and-whistles for `by-uuid` and `by-path` are unnecessary. The strategy to introduce these new features must be decided by the maintainers - I would go in incremental steps ;)

**Review by droidmonkey — APPROVED** [review]

**Comment by droidmonkey** [comment]
I cleaned up the code a bit, ready to merge once CI completes

**Comment by marquardts** [comment]
> [...] ready to merge once CI completes

There seems to be an issue with the `linux` agents. Apparently, TeamCity doesn't find the proper labels.

**Pull request merged**
